powerauth-java-server/src/main/java/io/getlime/security/powerauth/app/server/service/behavior/tasks/v3/ActivationServiceBehavior.java

```diff
@@ -300,27 +300,37 @@ public LookupActivationsResponse lookupActivations(List<String> userIds, List<Lo
         }
         List<ActivationRecordEntity> activationsList = activationRepository.lookupActivations(userIds, applicationIds, timestampLastUsedBefore, timestampLastUsedAfter, statuses);
 
-        if (activationsList != null) {
-            for (ActivationRecordEntity activation : activationsList) {
-                // Map between database object and service objects
-                LookupActivationsResponse.Activations activationServiceItem = new LookupActivationsResponse.Activations();
-                activationServiceItem.setActivationId(activation.getActivationId());
-                activationServiceItem.setActivationStatus(activationStatusConverter.convert(activation.getActivationStatus()));
-                activationServiceItem.setBlockedReason(activation.getBlockedReason());
-                activationServiceItem.setActivationName(activation.getActivationName());
-                activationServiceItem.setExtras(activation.getExtras());
-                activationServiceItem.setPlatform(activation.getPlatform());
-                activationServiceItem.setDeviceInfo(activation.getDeviceInfo());
-                activationServiceItem.setTimestampCreated(XMLGregorianCalendarConverter.convertFrom(activation.getTimestampCreated()));
-                activationServiceItem.setTimestampLastUsed(XMLGregorianCalendarConverter.convertFrom(activation.getTimestampLastUsed()));
-                activationServiceItem.setTimestampLastChange(XMLGregorianCalendarConverter.convertFrom(activation.getTimestampLastChange()));
-                activationServiceItem.setUserId(activation.getUserId());
-                activationServiceItem.setApplicationId(activation.getApplication().getId());
-                activationServiceItem.setApplicationName(activation.getApplication().getName());
-                // Unknown version is converted to 0 in SOAP
-                activationServiceItem.setVersion(activation.getVersion() == null ? 0L : activation.getVersion());
-                response.getActivations().add(activationServiceItem);
-            }
+        List<ActivationRecordEntity> filteredActivationList = new ArrayList<>();
+        // Filter activation by activation flags in case they are specified
+        if (activationFlags != null && !activationFlags.isEmpty()) {
+            List<ActivationRecordEntity> activationsWithFlags = activationsList.stream().filter(activation ->
```

Comment:
Are we sure that `activationList` returned from `.lookupActivations(..)` is always non-null?

Reply:
Yes, the repository returns an empty list in case no match is found for the query criteria.

Reply:
It is a good point, we can skip all this processing in case the list is empty.